python/src/iceberg/utils/schema_conversion.py

```diff
@@ -0,0 +1,451 @@
+            ...      }],
+            ... }
+            >>> actual = AvroSchemaConversion()._convert_record_type(record_type)
+            >>> expected = StructType(
+            ...     fields=(
+            ...         NestedField(
+            ...             field_id=509,
+            ...             name="contains_null",
+            ...             field_type=BooleanType(),
+            ...             is_optional=False,
+            ...             doc="True if any file has a null partition value",
+            ...         ),
+            ...         NestedField(
+            ...             field_id=518,
+            ...             name="contains_nan",
+            ...             field_type=BooleanType(),
+            ...             is_optional=True,
+            ...             doc="True if any file has a nan partition value",
+            ...         ),
+            ...     )
+            ... )
+            >>> expected == actual
+            True
+
+        Args:
+            record_type: The record type itself
+
+        Returns:
+        """
+        return StructType(*[self._convert_field(field) for field in record_type["fields"]])
+
+    def _convert_array_type(self, array_type: Dict[str, Any]) -> ListType:
+        element_type, element_is_optional = self._resolve_inner_type(array_type, "items", "element-id")
+        return ListType(element_id=array_type["element-id"], element_type=element_type, element_is_optional=element_is_optional)
+
+    def _convert_map_type(self, map_type: Dict[str, Any]) -> MapType:
+        """
+        Args:
+            map_type: The dict that describes the Avro map type
+
+        Examples:
+            >>> from iceberg.utils.schema_conversion import AvroSchemaConversion
+            >>> avro_field = {
+            ...     "type": "map",
+            ...     "values": ["long", "null"],
+            ...     "key-id": 101,
+            ...     "value-id": 102,
+            ... }
+            >>> actual = AvroSchemaConversion()._convert_map_type(avro_field)
+            >>> expected = MapType(
+            ...     key_id=101,
+            ...     key_type=StringType(),
+            ...     value_id=102,
+            ...     value_type=LongType(),
+            ...     value_is_optional=True
+            ... )
+            >>> actual == expected
+            True
+
+        Returns: A MapType
+        """
+        value_type, value_is_optional = self._resolve_union(map_type["values"])
+        return MapType(
+            key_id=map_type["key-id"],
+            # Avro only supports string keys
+            key_type=StringType(),
+            value_id=map_type["value-id"],
+            value_type=self._convert_schema(value_type),
+            value_is_optional=value_is_optional,
+        )
+
+    def _convert_logical_type(self, avro_logical_type: Dict[str, Any]) -> IcebergType:
+        """
+        When a logical type is found, we'll resolve it here. For the decimal and map
```

Comment:
Style: Avoid the use of personal pronouns, like "I" and "we". Those make docs longer and don't add clarity. Here, you can use "Convert a schema with a logical type annotation", which is more direct.

Reply:
Good point, thanks!